operate/qa/integration-tests/src/test/java/io/camunda/operate/elasticsearch/plugin/CustomStaticHeaderAddedFromPluginIT.java

```diff
@@ -0,0 +1,85 @@
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.operate.elasticsearch.plugin;
+
+import static io.camunda.operate.elasticsearch.plugin.CustomStaticHeaderAddedFromPluginIT.PLUGIN_CLASS_NAME;
+import static io.camunda.operate.elasticsearch.plugin.CustomStaticHeaderAddedFromPluginIT.PLUGIN_ID;
+import static io.camunda.operate.elasticsearch.plugin.CustomStaticHeaderAddedFromPluginIT.PLUGIN_JAR_BASE_PATH;
+import static io.camunda.operate.elasticsearch.plugin.CustomStaticHeaderAddedFromPluginIT.PLUGIN_JAR_FILE_NAME;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.transport.rest_client.RestClientTransport;
+import io.camunda.operate.elasticsearch.plugin.util.TestStaticCustomHeaderInterceptor;
+import io.camunda.operate.property.OperateProperties;
+import io.camunda.operate.util.TestApplication;
+import org.apache.http.HttpHost;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestWrapper;
+import org.apache.http.protocol.BasicHttpContext;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest(
+    classes = {TestApplication.class},
+    properties = {
+      OperateProperties.PREFIX + ".importer.startLoadingDataOnStartup = false",
+      OperateProperties.PREFIX + ".archiver.rolloverEnabled = false",
+      "spring.mvc.pathmatch.matching-strategy=ANT_PATH_MATCHER",
+      OperateProperties.PREFIX + ".multiTenancy.enabled = false",
+      "camunda.operate.zeebeElasticsearch.interceptor-plugins[0].id=" + PLUGIN_ID,
+      "camunda.operate.zeebeElasticsearch.interceptor-plugins[0].class-name=" + PLUGIN_CLASS_NAME,
+      "camunda.operate.zeebeElasticsearch.interceptor-plugins[0].jar-path="
+          + PLUGIN_JAR_BASE_PATH
+          + PLUGIN_JAR_FILE_NAME,
+      "camunda.operate.elasticsearch.interceptor-plugins[0].id=" + PLUGIN_ID,
+      "camunda.operate.elasticsearch.interceptor-plugins[0].class-name=" + PLUGIN_CLASS_NAME,
+      "camunda.operate.elasticsearch.interceptor-plugins[0].jar-path="
+          + PLUGIN_JAR_BASE_PATH
+          + PLUGIN_JAR_FILE_NAME
+    })
+public class CustomStaticHeaderAddedFromPluginIT extends PluginOperateSearchAbstractIT {
+
+  public static final String PLUGIN_ID = "my-plugin";
+  public static final String PLUGIN_CLASS_NAME =
+      "io.camunda.operate.elasticsearch.plugin.util.TestStaticCustomHeaderInterceptor";
+  public static final String PLUGIN_JAR_BASE_PATH = "/tmp/test01/";
+
+  @Autowired private ElasticsearchClient client;
+
+  @BeforeAll
+  @Override
+  public void beforeAllSetup() {
+    jarFile =
+        createCustomHeaderInterceptorJar(
+            PLUGIN_JAR_BASE_PATH, TestStaticCustomHeaderInterceptor.class);
+    jarFile.deleteOnExit();
```

Comment:
In these tests I am testing HTTP requests / interceptors. I cannot do it without loading the context because interceptor is registered in spring configuration.

Reply:
Moved to unit tests